<html>

  <head>
    <title>
      samtools
    </title>
  </head>

  <body bgcolor="#EEEEEE" link="#CC0000" alink="#FF3300" vlink="#000055">

  <article id="post-124" class="post-124 page type-page status-publish hentry">

  <header class="entry-header">
    <h1 class="entry-title">
      samtools
    </h1>
  </header>

  <!-- .entry-header -->
  <div class="entry-content">

  <h2>
    Introduction:
  </h2>

  <p>
    <b>SAMTOOLS</b> is a set of tools for high throughput sequencing data.
  </p>

  <p>
    SAMTOOLS is associated with the SAM (Sequence Alignment/Map) format,
    designed to store large nucleotide sequence alignments, and
    <ul>
      <li>
        can store information generated by various alignment programs;
      </li>
      <li>
        is simple enough to be easily generated, or converted;
      </li>
      <li>
        allows most operations on alignment to be done in a streaming fashion;
      </li>
      <li>
        allows the file to be indexed by genomic position;
      </li>
    </ul>
  </p>

  <p>
    SAMTOOLS is a set of utilities that manipulate alignments in the BAM
    format.  It imports from and exports to the SAM format, does sorting,
    merging and indexing, and allows to retrieve reads in any regions swiftly.
  </p>

  <p>
    SAMTOOLS is designed to work on a stream.  It regards an input file 
    '-' as the standard input (stdin) and an output file '-' as the 
    standard output (stdout).  Several commands can thus be combined with
    Unix pipes.  SAMTOOLS always outputs warning and error messages to the
    standard error output (stderr).
  </p>

  <p>
    SAMTOOLS is also able to open a BAM (but not SAM) file on a remote 
    FTP or HTTP server if the BAM file name starts with 'ftp://' or 
    'http://'.  SAMTOOLS checks the current working directory for the 
    index file and will download the index upon absence.  SAMTOOLS does  
    not retrieve the entire alignment file unless it is asked to do so.
  </p>

  <h2>
    Web Site:
  </h2>

  <p>
    The SAMTOOLS home page at sourceforge:<br>
    <a href = "http://samtools.sourceforge.net/">
               http://samtools.sourceforge.net/ </a>
  </p>

  <h2>
    Reference:
  </h2>

  <p>
    <ul>
      <li>
        SAMTOOLS documentation page at htslib.org:<br>
        <a href = "http://www.htslib.org/doc/samtools.html">
                   http://www.htslib.org/doc/samtools.html</a>,
      </li>
      <li>
        Heng Li, Bob Handsaker, Alec Wysoker, Tim Fennell, Jue Ruan, 
        Nils Homer, Gabor Marth, Goncalo Abecasis, Richard Durbin, 
        the 1000 Genome Project Data Processing Subgroup,<br>
        The Sequence alignment/map (SAM) format and SAMtools,<br>
        Bioinformatics,<br>
        Volume 25, pages 2078-2079, 2009.
      </li>
    </ul>
  </p>

  <h2>
    Usage:
  </h2>    

  <p>
    On any ARC cluster, check the installation details
    by typing "module spider samtools".
  </p> 

  <p>
    SAMTOOLS requires that the appropriate modules be loaded before it can 
    be used.  One version of the appropriate commands for use on NewRiver is:
    <pre>
module purge
module load gcc/5.2.0
module load samtools/1.2
    </pre>
  </p>

  <h2>
    Examples:
  </h2>

  <p>
    The following batch file converts a SAM file to BAN format:
<pre>
#! /bin/bash
#
#PBS -l walltime=0:05:00
#PBS -l nodes=1:ppn=1
#PBS -W group_list=newriver
#PBS -q open_q
#PBS -j oe
#
cd $PBS_O_WORKDIR
#
module purge
module load gcc/5.2.0
module load samtools/1.2
#
#  Convert a SAM file to BAM format.
#  * -b indicates the output should be in BAM format;
#  * -S indicates the input is in SAM format;
#  * -o names the output file.
#
samtools view -b -S -o sim_reads_aligned.bam sim_reads_aligned.sam
</pre>
  </p>

  <p>
    A complete set of files to carry out a similar process are available in
    <a href = "https://secure.hosting.vt.edu/www.arc.vt.edu/wp-content/uploads/2017/07/samtools_example.tar.gz">a tar file</a> 
  </p>

  </div><!-- .entry-content -->
</article><!-- #post-## -->

  </body>

</html>
